node-metrics/src/service/client_state/mod.rs

```diff
@@ -1359,6 +1359,8 @@ pub mod tests {
                 voters_1, voters_2
             ]))),
         );
+
+        drop(process_internal_client_message_handle)
```

Comment:
Shouldn't it be dropped when it goes out of scope anyway?

Reply:
Yeah it should. The test was failing with an error like stream has ended. I changed it to explicitly drop it, but this might be unrelated

Reply:
I did debug this test but didn't find anything

Reply:
I'll  try running the test

Reply:
the exact error was `InternalClientMessageProcessingTask stream closed, unable to process new requests from clients.`
cc @Ayiga

Reply:
@imabdulbasit doesn't fail for me without the drop. I don't mind adding it but I think it's a good idea to leave a comment as to why it was added because it looks a bit out of place.

Reply:
@sveitser yeah I couldn't replicate it either locally. I'll revert this change and see if CI passes. I think its a flaky test

Reply:
Oh! Actually, this might be erroring occasionally due to a clean up of the underlying channels. If the thread scheduling happens in an inconvenient order, then the channels will drop first, and that will cause process_internal_client_message_handle to fail reading from the channel which will result in a panic! as the task is a critical task for the node-validator's continued operation.

I think the better thing to do, rather than drop(process_internal_client_message_handle) is to just cancel the task and await it. This will ensure that it doesn't get scheduled anymore and will avoid the panic entirely.

It seems like a hacky work-around, but I'm not sure how else to maintain the duality of, "this is a critical task and should panic when running", but it is not possible to catch in testing without a specialized entry point to help tie into the panic.

Reply:
https://github.com/EspressoSystems/espresso-sequencer/issues/1862